lang/build.gradle

```diff
@@ -62,16 +62,70 @@ tasks.register('javadocJar', Jar) {
 
 publishing {
     publications {
-        create('maven', MavenPublication) {
-            artifactId = 'partiql-lang-kotlin'
+        maven(MavenPublication) {
+            // group id and version are set in the root build.gradle for all projects.
+            artifactId = "partiql-lang-kotlin"
+
+            from components.java
+            artifact sourcesJar
+            artifact javadocJar
 
-            from(components['java'])
-            artifact(tasks['sourcesJar'])
-            artifact(tasks['javadocJar'])
+            pom {
+                name = "PartiQL-Lang-Kotlin"
+                packaging = "jar"
+                url = "https://github.com/amzn/ion-schema-kotlin"
+                description = "An implementation of PartiQL for the JVM written in Kotlin."
+                scm {
+                    connection = "scm:git@github.com:partiql/partiql-lang-kotlin.git"
+                    developerConnection = "scm:git@github.com:partiql/partiql-lang-kotlin.git"
+                    url = "git@github.com:partiql/partiql-lang-kotlin.git"
+                }
+                licenses {
+                    license {
+                        name = "The Apache License, Version 2.0"
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                    }
+                }
+                developers {
+                    developer {
+                        name = "Amazon Ion Team"
+                        email = "partiql-community@amazon.com"
+                        organization = "PartiQL"
+                        organizationUrl = "https://github.com/partiql"
+                    }
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            url "https://oss.sonatype.org/service/local/staging/deploy/maven2"
+            credentials {
+                username ossrhUsername
+                password ossrhPassword
+            }
         }
     }
 }
 
+// The below enables developers and CI servers who do not have the GPG keys needed for signing to build locally,
+// while also signing release builds.
+// Signatures are only required when publishing to maven central anyway. We may need a more sophisticated means of
+// determining if we should sign the build artifacts someday, but this should suffice for now.
+if(!version.endsWith("SNAPSHOT")) {
```

Comment:
don't understand why do you need this. AFAIK signing plugin will only try to sign the maven publication. It will ask for the variables you defined in a `gradle.properties` but you can have it in the repo with dummy data, example: https://github.com/amzn/ion-schema-kotlin/blob/master/gradle.properties

Reply:
I don't think we should temporarily add secrets to any file where they might accidentally be committed and pushed.  I think we should use an [out-of-tree](https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties) `gradle.properties` instead to avoid such accidents.  

Leaving a non-functional template checked into the source tree feels like a rake in the leaves--you can remember its there but you might forget and then step on it.

I discovered the need for this when I had a copy of the exact `gradle.properties` file you linked and I executed:

`./gradlew publishToMavenLocal`

And it was refusing to publish because it couldn't access the signing key.  This works around that.  I'm open to other workarounds, if there are any.  The `ion-schema-kotlin` gradle build has the same problem, BTW.

Reply:
1. Don't see why we would add secrets to the working tree, take a look at the deployment process I linked you it uses a `gradle.properties` out of the working tree. The one in the working tree is there only as a placeholder values for gradle since replacing variables happens independent of the sign plugin.

If we ever want to publish `-SNAPSHOT` versions they would need to be signed, I don't see a reason to generate unsigned maven artifacts tbh so I'm OK with `./gradlew publishToMavenLocal` not working unless you have all things in place for the sign plugin to execute.

Reply:
I think having `publishToMavenLocal` working is important for anyone who wants to test their changes to PartiQL in other projects.  I will look around to see if there's another way avoid signing when publishing to the local repository.

Reply:
Your local m2 shouldn't really be part of your development loop, is too cumbersome. Both maven and gradle let you depend directly on a project or jar in your file system which is much more practical than having to build the jar and push to local m2.

I also think there is value in normalizing all our release process so we don't need to keep specific documentation. Both ion-schema and ion-java-path-extraction use gradle.properties with dummy values